airbyte-integrations/connectors/source-s3/poetry.lock

```diff
@@ -865,13 +876,13 @@ jsonpointer = ">=1.9"
 
 [[package]]
 name = "jsonpointer"
-version = "2.4"
+version = "3.0.0"
```

Comment:
This major change not related to pydantic got me worried a bit. The path to this dependency is: airbyte-cdk -> langchain-core -> jsonpatch -> jsonpointer

Given we don't use langchain-core in source-s3, this should be fine

Reply:
this is actually a good question - im not sure why langchain is being installed in any case 🤔

Reply:
I also don't know how it works. It is [part of the optional dependencies](https://github.com/airbytehq/airbyte/blob/48aa409d04047bc82e63d43e0271a95c7b3a9a1d/airbyte-cdk/python/pyproject.toml#L50). Is poetry installing optional things as well? @alafanechere maybe? 

In any case, I don't think this is a blocker as we should not use this as part of source-s3

Reply:
Cool, I don't think so either, I was also looking for augustin for help though :D

Reply:
I do think the default poetry install command installs all declared extras

Reply:
We specify extras that we want to install, but we install file-based and not vector-db-based

Reply:
maybe its the `--no-dev` that we need to add to not get it to install everything